File: week 6/readability/readability.py
from cs50 import get_string


def main():
    s = get_string("Text: ")

    letters = 0
    word = 0
    sentences = 0
    n = len(s)

    if s[0].isalnum() == True:
        word += 1
# All 3 cases for letters,words and sentences
    for i in range(n):
        if s[i].isalpha() == True:
            letters = letters + 1
        if ((s[i].isspace() == True) and ((s[i+1].isalnum() == True) or s[i+1] == '"')):
            word += 1
        if (s[i] == '!' or s[i] == '?' or s[i] == '.') and (s[i-1].isalnum() == True):
            sentences += 1

# Coleman-lian index
    grade = 0.0588*(100.0*letters/word) - 0.296*(100.0*sentences/word) - 15.8

# round the result
    num = round(grade)

    if num <= 1:
        print("Before Grade 1\n")
    elif num > 16:
        print("Grade 16+\n")
    else:
        print("Grade", str(num))


if __name__ == "__main__":
    main()

File: week 2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int letter;
int word;
int sentence;


int main(void)
{

// prompt the user with the question

    string article = get_string("What's the article?: ");

// set the length of article

    int n = strlen(article);

// add +1 if the article starts with alphanumeric letter

    if (isalnum(article[0]))
    {
        word += 1;
    }

// count words

    for (int i = 0; i < n;  i++)
    {
        // count letters

        if (isalnum(article[i]))
        {
            letter++;
        }

        // count words

        if (isspace(article[i]) && (isalnum(article[i + 1]) || article[i + 1] == '"'))
        {
            word++;
        }

        // count sentences

        if ((article[i] == '!' || article[i] == '?' || article[i] == '.') && isalnum(article[i - 1]))
        {
            sentence++;
        }

    }

// calculate Coleman-Liau index

    double grade = 0.0588 * (100.0 * (double)letter / word) - 0.296 * (100.0 * (double)sentence / word) - 15.8;

// debugger
    int brad = (int) round(grade);
    //  printf("Letters: %lf\n Words: %i\n Sentences: %lf\n", (double)letter, word, (double)sentence);

// print result
    if (brad <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (brad < 16)
    {
        printf("Grade %d\n", brad);
    }
    else
    {
        printf("Grade 16+\n");
    }
}